In [36]:
import pandas as pd
import numpy as np

In [37]:
df_players = pd.read_csv('Series Stats/Player Game by Game_Series Results_Table.csv')
df_teams = pd.read_csv('Game by Game Team Stats/Game by Game Results_Series Results_Table.csv')


In [64]:
# Left join based on the "series ID" column
merged_left = pd.merge(df_players, df_teams, on=['Team', 'Series ID'], how="left")
merged_left['Win/Loss'] = np.where(merged_left['Maps Won'] > merged_left['Maps Lost'], 'Win', 'Loss')
merged_left

Player   Team Opponent_x        Date_x  Stage_x  Series ID  Kills  \
0       Pred  Optic       Faze  Jan 28, 2024  Stage 1       59.0     73   
1      Dashy  Optic       Faze  Jan 28, 2024  Stage 1       59.0     54   
2    Shotzzy  Optic       Faze  Jan 28, 2024  Stage 1       59.0     51   
3      aBeZy   Faze      Optic  Jan 28, 2024  Stage 1       59.0     71   
4     Drazah   Faze      Optic  Jan 28, 2024  Stage 1       59.0     61   
..       ...    ...        ...           ...      ...        ...    ...   
475   Drazah   Faze      Ultra  Jan 28, 2024      NaN        NaN     17   
476   CleanX  Ultra       Faze  Jan 28, 2024      NaN        NaN     22   
477  Insight  Ultra       Faze  Jan 28, 2024      NaN        NaN     12   
478    Envoy  Ultra       Faze  Jan 28, 2024      NaN        NaN     23   
479    Scrap  Ultra       Faze  Jan 28, 2024      NaN        NaN     28   

     Deaths   K/D % of Teams Kills % of Teams Deaths        Date_y  Stage_y  \
0        66  1.11           31.88%            24.26%  Jan 28, 2024  Stage 1   
1        61  0.89           23.58%            22.43%  Jan 28, 2024  Stage 1   
2        72  0.71           22.27%            26.47%  Jan 28, 2024  Stage 1   
3        59  1.20            26.1%            25.76%  Jan 28, 2024  Stage 1   
4        61  1.00           22.43%            26.64%  Jan 28, 2024  Stage 1   
..      ...   ...              ...               ...           ...      ...   
475      24  0.71           23.94%            28.24%  Jan 28, 2024      NaN   
476      18  1.22           25.88%            25.35%  Jan 28, 2024      NaN   
477      16  0.75           14.12%            22.54%  Jan 28, 2024      NaN   
478      19  1.21           27.06%            26.76%  Jan 28, 2024      NaN   
479      18  1.56           32.94%            25.35%  Jan 28, 2024      NaN   

    Opponent_y  Maps Won  Maps Lost  Net Kills Win/Loss  
0         Faze         1          3        -43     Loss  
1         Faze         1          3        -43     Loss  
2         Faze         1          3        -43     Loss  
3        Optic         3          1         43      Win  
4        Optic         3          1         43      Win  
..         ...       ...        ...        ...      ...  
475      Ultra         0          1        -14     Loss  
476       Faze         1          0         14      Win  
477       Faze         1          0         14      Win  
478       Faze         1          0         14      Win  
479       Faze         1          0         14      Win  

[480 rows x 18 columns]

What I did above was just merge two tables together. One being a players stats for every game they played in every series. The other table is a Team stats table for each series with some overall info on the series per team. I merged these two so that we can see how each player preformed in each series and how that effected the outcome. To see if there are any players that carry a heavier weight on the teams score when they play good or bad. I also added a Win / Loss column to easily show if the team won the series or lost.

The next analysis I want to do is examine each player on Optic to see how their perfomance effects the teams outcome

In [39]:
# We will start with Shotzzy
shotzzy_df = merged_left[merged_left['Player']=='Shotzzy']
shotzzy_df

Player   Team    Opponent_x        Date_x  Stage_x  Series ID  Kills  \
2    Shotzzy  Optic          Faze  Jan 28, 2024  Stage 1       59.0     51   
18   Shotzzy  Optic         Rokkr  Jan 28, 2024  Stage 1       57.0     66   
32   Shotzzy  Optic         Surge  Jan 27, 2024  Stage 1       55.0     68   
54   Shotzzy  Optic          Faze  Jan 27, 2024  Stage 1       53.0     76   
110  Shotzzy  Optic      Heretics  Jan 25, 2024  Stage 1       46.0     59   
140  Shotzzy  Optic          Faze  Jan 21, 2024  Stage 1       42.0     42   
206  Shotzzy  Optic      Heretics  Jan 19, 2024  Stage 1       34.0     41   
254  Shotzzy  Optic        Legion  Jan 13, 2024  Stage 1       28.0     63   
326  Shotzzy  Optic  Royal Ravens  Dec 17, 2023  Stage 1       19.0     46   
366  Shotzzy  Optic         Surge  Dec 15, 2023  Stage 1       14.0     52   
390  Shotzzy  Optic         Rokkr  Dec 10, 2023  Stage 1       11.0     82   
454  Shotzzy  Optic     Subliners   Dec 8, 2023  Stage 1        3.0     86   

     Deaths   K/D % of Teams Kills % of Teams Deaths        Date_y  Stage_y  \
2        72  0.71           22.27%            26.47%  Jan 28, 2024  Stage 1   
18       66  1.00           23.83%            26.83%  Jan 28, 2024  Stage 1   
32       69  0.99           21.59%            26.95%  Jan 27, 2024  Stage 1   
54       91  0.84           24.52%            27.08%  Jan 27, 2024  Stage 1   
110      39  1.51           28.92%            23.49%  Jan 25, 2024  Stage 1   
140      56  0.75           21.76%            24.89%  Jan 21, 2024  Stage 1   
206      44  0.93           26.97%            25.14%  Jan 19, 2024  Stage 1   
254      70  0.90           23.42%            26.52%  Jan 13, 2024  Stage 1   
326      34  1.35           25.14%            28.33%  Dec 17, 2023  Stage 1   
366      63  0.83            19.4%             25.3%  Dec 15, 2023  Stage 1   
390      69  1.19           27.61%            27.49%  Dec 10, 2023  Stage 1   
454      85  1.01           26.54%            26.81%   Dec 8, 2023  Stage 1   

       Opponent_y  Maps Won  Maps Lost  Net Kills Win/Loss  
2            Faze         1          3        -43     Loss  
18          Rokkr         3          1         31      Win  
32          Surge         3          1         59      Win  
54           Faze         2          3        -26     Loss  
110      Heretics         3          0         38      Win  
140          Faze         0          3        -32     Loss  
206      Heretics         0          3        -23     Loss  
254        Legion         3          2          5      Win  
326  Royal Ravens         3          0         63      Win  
366         Surge         3          1         19      Win  
390         Rokkr         3          1         46      Win  
454     Subliners         2          3          7     Loss

In [46]:
# This is just getting the league average of KD across Major 1
average_kd = merged_left['K/D'].mean()
average_kd

1.0176041666666666

In [90]:
average_kills_series = merged_left['Kills'].mean()
average_kills_series

62.50208333333333

In [65]:
# This code is simply removing the % from the columns to be able to convert to numeric value
merged_left['% of Teams Kills'] = merged_left['% of Teams Kills'].str.replace('%','')
merged_left['% of Teams Deaths'] = merged_left['% of Teams Deaths'].str.replace('%','')
# This code converts the columns to numeric values to be able to take the average 
merged_left['% of Teams Kills'] = pd.to_numeric(merged_left['% of Teams Kills'])
merged_left['% of Teams Deaths'] = pd.to_numeric(merged_left['% of Teams Deaths'])

In [66]:
average_team_kills = merged_left['% of Teams Kills'].mean()
average_team_kills


25.000062500000002

In [67]:
average_team_deaths = merged_left['% of Teams Deaths'].mean()
average_team_deaths

24.999874999999996

In [41]:
shotzzy_df[shotzzy_df['K/D']>= average_kd]['K/D'].count()

3

In [42]:
shotzzy_df[shotzzy_df['K/D']< average_kd]['K/D'].count()

9

Based off of the average KD we calculated across the matches from Major 1, we can see that Shotzzy had an above average KD in 3/12 series. This of course, is not ideal, but lets dig a little deeper to see what games he had an above or below average KD, if the team won, who they played, etc.

In [45]:
shotzzy_df.sort_values('K/D', ascending=False)

Player   Team    Opponent_x        Date_x  Stage_x  Series ID  Kills  \
110  Shotzzy  Optic      Heretics  Jan 25, 2024  Stage 1       46.0     59   
326  Shotzzy  Optic  Royal Ravens  Dec 17, 2023  Stage 1       19.0     46   
390  Shotzzy  Optic         Rokkr  Dec 10, 2023  Stage 1       11.0     82   
454  Shotzzy  Optic     Subliners   Dec 8, 2023  Stage 1        3.0     86   
18   Shotzzy  Optic         Rokkr  Jan 28, 2024  Stage 1       57.0     66   
32   Shotzzy  Optic         Surge  Jan 27, 2024  Stage 1       55.0     68   
206  Shotzzy  Optic      Heretics  Jan 19, 2024  Stage 1       34.0     41   
254  Shotzzy  Optic        Legion  Jan 13, 2024  Stage 1       28.0     63   
54   Shotzzy  Optic          Faze  Jan 27, 2024  Stage 1       53.0     76   
366  Shotzzy  Optic         Surge  Dec 15, 2023  Stage 1       14.0     52   
140  Shotzzy  Optic          Faze  Jan 21, 2024  Stage 1       42.0     42   
2    Shotzzy  Optic          Faze  Jan 28, 2024  Stage 1       59.0     51   

     Deaths   K/D % of Teams Kills % of Teams Deaths        Date_y  Stage_y  \
110      39  1.51           28.92%            23.49%  Jan 25, 2024  Stage 1   
326      34  1.35           25.14%            28.33%  Dec 17, 2023  Stage 1   
390      69  1.19           27.61%            27.49%  Dec 10, 2023  Stage 1   
454      85  1.01           26.54%            26.81%   Dec 8, 2023  Stage 1   
18       66  1.00           23.83%            26.83%  Jan 28, 2024  Stage 1   
32       69  0.99           21.59%            26.95%  Jan 27, 2024  Stage 1   
206      44  0.93           26.97%            25.14%  Jan 19, 2024  Stage 1   
254      70  0.90           23.42%            26.52%  Jan 13, 2024  Stage 1   
54       91  0.84           24.52%            27.08%  Jan 27, 2024  Stage 1   
366      63  0.83            19.4%             25.3%  Dec 15, 2023  Stage 1   
140      56  0.75           21.76%            24.89%  Jan 21, 2024  Stage 1   
2        72  0.71           22.27%            26.47%  Jan 28, 2024  Stage 1   

       Opponent_y  Maps Won  Maps Lost  Net Kills Win/Loss  
110      Heretics         3          0         38      Win  
326  Royal Ravens         3          0         63      Win  
390         Rokkr         3          1         46      Win  
454     Subliners         2          3          7     Loss  
18          Rokkr         3          1         31      Win  
32          Surge         3          1         59      Win  
206      Heretics         0          3        -23     Loss  
254        Legion         3          2          5      Win  
54           Faze         2          3        -26     Loss  
366         Surge         3          1         19      Win  
140          Faze         0          3        -32     Loss  
2            Faze         1          3        -43     Loss

In [48]:
shotzzy_df[shotzzy_df['Win/Loss']=='Win']['Win/Loss'].count()

7

After sorting the table my Shotzzys KD and getting the count of Wins that Optic had across Major 1 we can see some interesting things. 

1) The first finding is that every series Shotzzy went over the average KD, the team won. And not just any win but two 3-0 wins and a 3-1 win. In addition, Optic is 4-1 and when he has at least a 1 KD. And only 2-4 when he goes below a 1 KD. It is safe to say when Shotzzy is playing well the team is winning.
2) His kryptonite, as of Major 1, is Faze. Three of his 4 worst series come against Faze. All of which include losses.
3)  Final point is just the difference on LAN an Online. Jan 25-28th was on LAN. Dec 8-Jan 21st was Online. There is not any real noticeable difference in his play from online to LAN.

Next lets take a look at the other returning Optic player from last season, Dashy

In [69]:
dashy_df = merged_left[merged_left['Player']=='Dashy']
dashy_df

Player   Team    Opponent_x        Date_x  Stage_x  Series ID  Kills  \
1    Dashy  Optic          Faze  Jan 28, 2024  Stage 1       59.0     54   
17   Dashy  Optic         Rokkr  Jan 28, 2024  Stage 1       57.0     64   
35   Dashy  Optic         Surge  Jan 27, 2024  Stage 1       55.0     75   
55   Dashy  Optic          Faze  Jan 27, 2024  Stage 1       53.0     58   
108  Dashy  Optic      Heretics  Jan 25, 2024  Stage 1       46.0     41   
139  Dashy  Optic          Faze  Jan 21, 2024  Stage 1       42.0     47   
205  Dashy  Optic      Heretics  Jan 19, 2024  Stage 1       34.0     42   
252  Dashy  Optic        Legion  Jan 13, 2024  Stage 1       28.0     60   
324  Dashy  Optic  Royal Ravens  Dec 17, 2023  Stage 1       19.0     46   
364  Dashy  Optic         Surge  Dec 15, 2023  Stage 1       14.0     64   
388  Dashy  Optic         Rokkr  Dec 10, 2023  Stage 1       11.0     72   
452  Dashy  Optic     Subliners   Dec 8, 2023  Stage 1        3.0     65   

     Deaths   K/D  % of Teams Kills  % of Teams Deaths        Date_y  Stage_y  \
1        61  0.89             23.58              22.43  Jan 28, 2024  Stage 1   
17       56  1.14             23.10              22.76  Jan 28, 2024  Stage 1   
35       55  1.36             23.81              21.48  Jan 27, 2024  Stage 1   
55       81  0.72             18.71              24.11  Jan 27, 2024  Stage 1   
108      31  1.32             20.10              18.67  Jan 25, 2024  Stage 1   
139      55  0.85             24.35              24.44  Jan 21, 2024  Stage 1   
205      47  0.89             27.63              26.86  Jan 19, 2024  Stage 1   
252      62  0.97             22.30              23.48  Jan 13, 2024  Stage 1   
324      25  1.84             25.14              20.83  Dec 17, 2023  Stage 1   
364      50  1.28             23.88              20.08  Dec 15, 2023  Stage 1   
388      57  1.26             24.24              22.71  Dec 10, 2023  Stage 1   
452      74  0.88             20.06              23.34   Dec 8, 2023  Stage 1   

       Opponent_y  Maps Won  Maps Lost  Net Kills Win/Loss  
1            Faze         1          3        -43     Loss  
17          Rokkr         3          1         31      Win  
35          Surge         3          1         59      Win  
55           Faze         2          3        -26     Loss  
108      Heretics         3          0         38      Win  
139          Faze         0          3        -32     Loss  
205      Heretics         0          3        -23     Loss  
252        Legion         3          2          5      Win  
324  Royal Ravens         3          0         63      Win  
364         Surge         3          1         19      Win  
388         Rokkr         3          1         46      Win  
452     Subliners         2          3          7     Loss

In [70]:
dashy_df[dashy_df['K/D']>=average_kd]['K/D'].count()

6

In [71]:
dashy_df[dashy_df['K/D']<average_kd]['K/D'].count()

6

Dashy had an above average KD in 6/12 matches throughout Major 1. With 2 of his top 3 performances coming on LAN.

In [72]:
dashy_df[(dashy_df['K/D']>=average_kd) & (dashy_df['Win/Loss']=='Win')]['Win/Loss'].count()

6

In [73]:
dashy_df.sort_values('K/D', ascending=False)

Player   Team    Opponent_x        Date_x  Stage_x  Series ID  Kills  \
324  Dashy  Optic  Royal Ravens  Dec 17, 2023  Stage 1       19.0     46   
35   Dashy  Optic         Surge  Jan 27, 2024  Stage 1       55.0     75   
108  Dashy  Optic      Heretics  Jan 25, 2024  Stage 1       46.0     41   
364  Dashy  Optic         Surge  Dec 15, 2023  Stage 1       14.0     64   
388  Dashy  Optic         Rokkr  Dec 10, 2023  Stage 1       11.0     72   
17   Dashy  Optic         Rokkr  Jan 28, 2024  Stage 1       57.0     64   
252  Dashy  Optic        Legion  Jan 13, 2024  Stage 1       28.0     60   
1    Dashy  Optic          Faze  Jan 28, 2024  Stage 1       59.0     54   
205  Dashy  Optic      Heretics  Jan 19, 2024  Stage 1       34.0     42   
452  Dashy  Optic     Subliners   Dec 8, 2023  Stage 1        3.0     65   
139  Dashy  Optic          Faze  Jan 21, 2024  Stage 1       42.0     47   
55   Dashy  Optic          Faze  Jan 27, 2024  Stage 1       53.0     58   

     Deaths   K/D  % of Teams Kills  % of Teams Deaths        Date_y  Stage_y  \
324      25  1.84             25.14              20.83  Dec 17, 2023  Stage 1   
35       55  1.36             23.81              21.48  Jan 27, 2024  Stage 1   
108      31  1.32             20.10              18.67  Jan 25, 2024  Stage 1   
364      50  1.28             23.88              20.08  Dec 15, 2023  Stage 1   
388      57  1.26             24.24              22.71  Dec 10, 2023  Stage 1   
17       56  1.14             23.10              22.76  Jan 28, 2024  Stage 1   
252      62  0.97             22.30              23.48  Jan 13, 2024  Stage 1   
1        61  0.89             23.58              22.43  Jan 28, 2024  Stage 1   
205      47  0.89             27.63              26.86  Jan 19, 2024  Stage 1   
452      74  0.88             20.06              23.34   Dec 8, 2023  Stage 1   
139      55  0.85             24.35              24.44  Jan 21, 2024  Stage 1   
55       81  0.72             18.71              24.11  Jan 27, 2024  Stage 1   

       Opponent_y  Maps Won  Maps Lost  Net Kills Win/Loss  
324  Royal Ravens         3          0         63      Win  
35          Surge         3          1         59      Win  
108      Heretics         3          0         38      Win  
364         Surge         3          1         19      Win  
388         Rokkr         3          1         46      Win  
17          Rokkr         3          1         31      Win  
252        Legion         3          2          5      Win  
1            Faze         1          3        -43     Loss  
205      Heretics         0          3        -23     Loss  
452     Subliners         2          3          7     Loss  
139          Faze         0          3        -32     Loss  
55           Faze         2          3        -26     Loss

My takeaways for Dashy from the data after analyzing is as follows:
1) Dashy is a MAJOR factor on if Optic wins matches or not. Optic is 6-0 when Dashy has an above average KD. With their closest series being a 3-1. When Dashy is performing below league average they are 1-5, with their only win in a game 5 victory against Vegas Legion.
2) Something we will more than likely see for all Optic players is their poor performance vs Faze in all three matchups so far this year. Like Shotzzy, Dashy's worst two series this year are vs Faze
3) With 3 out of his top 6 performances being on LAN and the other 3 online, I would conclude that he is pretty consistent across both. Do note that 3 of his 4 worst performances did come online, so when he is playing at his worst it is usually online.

Now lets check out Optic's biggest off-season pick up, Pred

In [75]:
pred_df = merged_left[merged_left['Player']=='Pred']
pred_df

Player   Team    Opponent_x        Date_x  Stage_x  Series ID  Kills  \
0     Pred  Optic          Faze  Jan 28, 2024  Stage 1       59.0     73   
22    Pred  Optic         Rokkr  Jan 28, 2024  Stage 1       57.0     81   
33    Pred  Optic         Surge  Jan 27, 2024  Stage 1       55.0     88   
52    Pred  Optic          Faze  Jan 27, 2024  Stage 1       53.0     92   
105   Pred  Optic      Heretics  Jan 25, 2024  Stage 1       46.0     51   
143   Pred  Optic          Faze  Jan 21, 2024  Stage 1       42.0     56   
200   Pred  Optic      Heretics  Jan 19, 2024  Stage 1       34.0     34   
253   Pred  Optic        Legion  Jan 13, 2024  Stage 1       28.0     68   
325   Pred  Optic  Royal Ravens  Dec 17, 2023  Stage 1       19.0     46   
365   Pred  Optic         Surge  Dec 15, 2023  Stage 1       14.0     83   
389   Pred  Optic         Rokkr  Dec 10, 2023  Stage 1       11.0     81   
453   Pred  Optic     Subliners   Dec 8, 2023  Stage 1        3.0     74   

     Deaths   K/D  % of Teams Kills  % of Teams Deaths        Date_y  Stage_y  \
0        66  1.11             31.88              24.26  Jan 28, 2024  Stage 1   
22       59  1.37             29.24              23.98  Jan 28, 2024  Stage 1   
33       67  1.31             27.94              26.17  Jan 27, 2024  Stage 1   
52       78  1.18             29.68              23.21  Jan 27, 2024  Stage 1   
105      44  1.16             25.00              26.51  Jan 25, 2024  Stage 1   
143      54  1.04             29.02              24.00  Jan 21, 2024  Stage 1   
200      44  0.77             22.37              25.14  Jan 19, 2024  Stage 1   
253      65  1.05             25.28              24.62  Jan 13, 2024  Stage 1   
325      32  1.44             25.14              26.67  Dec 17, 2023  Stage 1   
365      63  1.32             30.97              25.30  Dec 15, 2023  Stage 1   
389      60  1.35             27.27              23.90  Dec 10, 2023  Stage 1   
453      75  0.99             22.84              23.66   Dec 8, 2023  Stage 1   

       Opponent_y  Maps Won  Maps Lost  Net Kills Win/Loss  
0            Faze         1          3        -43     Loss  
22          Rokkr         3          1         31      Win  
33          Surge         3          1         59      Win  
52           Faze         2          3        -26     Loss  
105      Heretics         3          0         38      Win  
143          Faze         0          3        -32     Loss  
200      Heretics         0          3        -23     Loss  
253        Legion         3          2          5      Win  
325  Royal Ravens         3          0         63      Win  
365         Surge         3          1         19      Win  
389         Rokkr         3          1         46      Win  
453     Subliners         2          3          7     Loss

In [81]:
pred_df[pred_df['K/D']>average_kd]['K/D'].count()

10

In [82]:
pred_df[(pred_df['K/D']>average_kd) & (pred_df['Win/Loss']=='Win')]['Win/Loss'].count()

7

Okay so right off the bat we see a couple interesting things here. Pred went above the average KD mark in 10/12 games over Major 1, and they won 7 out of those 10 games. Who were the three losses to? You guessed it, Faze.

In [79]:
pred_df[pred_df['% of Teams Kills']>average_team_kills]['% of Teams Kills'].count()

9

In [80]:
pred_df[(pred_df['% of Teams Kills']>average_team_kills) & (pred_df['% of Teams Deaths']<average_team_deaths)]['Player'].count()

6

When analyzing Pred's overall stats I noticed he was having a lot of high kills series. So I dove into the % of the teams kills he had each series. In 9 of the 12 series he had over the league average of % of his teams kills, which is 25%. With a whopping 31.88% of his teams kills in their Losers bracket final matchup vs Faze in Major 1. That percentage is good for 9th best for the whole Major.

In [87]:
merged_left[(merged_left['Kills']>90) & (merged_left['Opponent_x']=='Faze')]

Player   Team Opponent_x        Date_x  Stage_x  Series ID  Kills  Deaths  \
13  Scrap  Ultra       Faze  Jan 28, 2024  Stage 1       58.0     97      90   
52   Pred  Optic       Faze  Jan 27, 2024  Stage 1       53.0     92      78   

     K/D  % of Teams Kills  % of Teams Deaths        Date_y  Stage_y  \
13  1.08             28.53              24.86  Jan 28, 2024  Stage 1   
52  1.18             29.68              23.21  Jan 27, 2024  Stage 1   

   Opponent_y  Maps Won  Maps Lost  Net Kills Win/Loss  
13       Faze         3          1        -22      Win  
52       Faze         2          3        -26     Loss

These are my biggest takeaways from Pred's Major 1:
1) Pred is only 1 of 2 plays the whole Major to drop over 90 kills on Faze; Scrap being the other one. This along with his high KD in all three matches against Faze show that, unlike Shotzzy and Dashy, he actually plays pretty well vs them
2) Pred having over 25% of his teams kills in 9/12 series in Major 1 shows just how big of an impact he is on the map. His stats and play throughout this Major is reminiscent of the retired Optic legend Scumpii. 
3) Another impressive stat I noticed was that Pred's lowest KD on LAN was 1.16. While he is still good online, his best performances come on LAN.
4) To close it out, Pred clearly plays a big role on this team, as long as he continues to play well and lead this team they should all follow suit behind him.

Okay now lets take a look at Optics other off-season pickup, Kenny

In [89]:
kenny_df = merged_left[merged_left['Player']=='Kenny']
kenny_df

Player   Team    Opponent_x        Date_x  Stage_x  Series ID  Kills  \
6    Kenny  Optic          Faze  Jan 28, 2024  Stage 1       59.0     51   
19   Kenny  Optic         Rokkr  Jan 28, 2024  Stage 1       57.0     66   
36   Kenny  Optic         Surge  Jan 27, 2024  Stage 1       55.0     84   
49   Kenny  Optic          Faze  Jan 27, 2024  Stage 1       53.0     84   
109  Kenny  Optic      Heretics  Jan 25, 2024  Stage 1       46.0     53   
142  Kenny  Optic          Faze  Jan 21, 2024  Stage 1       42.0     48   
207  Kenny  Optic      Heretics  Jan 19, 2024  Stage 1       34.0     35   
255  Kenny  Optic        Legion  Jan 13, 2024  Stage 1       28.0     78   
327  Kenny  Optic  Royal Ravens  Dec 17, 2023  Stage 1       19.0     45   
367  Kenny  Optic         Surge  Dec 15, 2023  Stage 1       14.0     69   
391  Kenny  Optic         Rokkr  Dec 10, 2023  Stage 1       11.0     62   
455  Kenny  Optic     Subliners   Dec 8, 2023  Stage 1        3.0     99   

     Deaths   K/D  % of Teams Kills  % of Teams Deaths        Date_y  Stage_y  \
6        73  0.70             22.27              26.84  Jan 28, 2024  Stage 1   
19       65  1.02             23.83              26.42  Jan 28, 2024  Stage 1   
36       65  1.29             26.67              25.39  Jan 27, 2024  Stage 1   
49       86  0.98             27.10              25.60  Jan 27, 2024  Stage 1   
109      52  1.02             25.98              31.33  Jan 25, 2024  Stage 1   
142      60  0.80             24.87              26.67  Jan 21, 2024  Stage 1   
207      40  0.88             23.03              22.86  Jan 19, 2024  Stage 1   
255      67  1.16             29.00              25.38  Jan 13, 2024  Stage 1   
327      29  1.55             24.59              24.17  Dec 17, 2023  Stage 1   
367      73  0.95             25.75              29.32  Dec 15, 2023  Stage 1   
391      65  0.95             20.88              25.90  Dec 10, 2023  Stage 1   
455      83  1.19             30.56              26.18   Dec 8, 2023  Stage 1   

       Opponent_y  Maps Won  Maps Lost  Net Kills Win/Loss  
6            Faze         1          3        -43     Loss  
19          Rokkr         3          1         31      Win  
36          Surge         3          1         59      Win  
49           Faze         2          3        -26     Loss  
109      Heretics         3          0         38      Win  
142          Faze         0          3        -32     Loss  
207      Heretics         0          3        -23     Loss  
255        Legion         3          2          5      Win  
327  Royal Ravens         3          0         63      Win  
367         Surge         3          1         19      Win  
391         Rokkr         3          1         46      Win  
455     Subliners         2          3          7     Loss

In [91]:
kenny_df[kenny_df['K/D']>average_kd]['Player'].count()

6

In [93]:
kenny_df[(kenny_df['K/D']>average_kd) & (kenny_df['Win/Loss']=='Win')]['Player'].count()

5

In [94]:
kenny_df[(kenny_df['K/D']<average_kd) & (kenny_df['Win/Loss']=='Win')]['Player'].count()

2

In [99]:
kenny_df[(kenny_df['K/D']<average_kd) & (kenny_df['Win/Loss']=='Loss')]['Player'].count()

4

In [98]:
kenny_df[kenny_df['% of Teams Deaths']> average_team_deaths]['Player'].count()

10

Recap of Kenny's Major 1:
1) Overall Kenny had a very average Major. Going above the average KD 6 times, the team won 5/6 of those times. This is pretty standard across the team, as in when a player does well they usually tend to win.
2) Following that up, one noticeable stat from Kenny was that he had over 25% of the teams deaths in 10 out of 12 matches in major 1. This could turn into a problem for Optic if he doesnt get this under control quickly. With the role he plays on the team, he should not be over a quarter of the teams deaths. This just goes back to winning those 1 on 1 gun fights a lot of the time.
3) Overall, not a terrible Major from Kenny, but he certainly needs to clean up his deaths going into the next Major. Every loss but one, he accounted for over 25% of the deaths for his team. This could turn out to be a pivotal reason on if Optic wins or loses going forward.

In [103]:
shotzzy_df[shotzzy_df['% of Teams Deaths']> average_team_deaths]['Player'].count()

10

In [104]:
dashy_df[dashy_df['% of Teams Deaths']> average_team_deaths]['Player'].count()

1

In [105]:
pred_df[pred_df['% of Teams Deaths']> average_team_deaths]['Player'].count()

5

WHOLE TEAM RECAP FOR MAJOR 1:
1) Without a doubt the two best players for Optic in Major 1 was Pred and Dashy. Kenny and Shotzzy had unusually poor performances in comparison to their two counterparts.
2) As you can see above, I went ahead and checked every players count of how many times they went over 25% of the teams deaths. Shottzy and Kenny both with 10, while Dashy and Pred have 1 and 5. This undoubtedly contributes to the individual players success, which in turn plays into how the team performs.
3) To wrap up, Optic had a good Major finishing 3rd overall. I think with a little better play from Kenny and Shotzzy this team could become a force to be reckoned with. I am looking forward to the adjustments they make going into Major 2